src/decision/location_selection.jl

```diff
@@ -0,0 +1,266 @@
+    n_ranks=length(ranks.sites),
+    agg_func=nothing,
+) where {D,T,A}
+    if !isnothing(agg_func)
+        return agg_func(ranks_to_frequencies(ranks, n_ranks))
+    end
+
+    return ranks_to_frequencies(ranks, n_ranks)
+end
+
+"""
+    location_selection_frequencies(ranks::NamedDimsArray;n_loc_int::Int64=5)
+    location_selection_frequencies(inv_log::NamedDimsArray{D,T,4,A};dims::Union{Symbol,Vector{Symbol}}=:coral_id) where {D,T,A}
+
+Post-processing function for intervention logs. Calculates the frequencies with which locations were selected for a particular intervention,
+for a selection of scenarios (e.g. selected robust scenarios).
+
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+- `n_loc_int` : number of locations which are intervened at for each intervention decision.
+- `dims` : dimensions to sum selection frequencies over.
+
+# Returns 
+Counts for location selection at each location in the domain.
+"""
+function location_selection_frequencies(
+    ranks::NamedDimsArray;
+    n_loc_int::Int64=5,
+)
+    ranks_frequencies = ranks_to_frequencies(ranks; n_ranks=n_loc_int)
+    loc_count = sum(ranks_frequencies[ranks=1:n_loc_int], dims=2)[ranks=1]
+
+    return loc_count
+end
+function location_selection_frequencies(
+    inv_log::NamedDimsArray{D,T,4,A};
+    dims::Union{Symbol,Vector{Symbol}}=:coral_id,
+) where {D,T,A}
+    loc_count = dropdims(
+        sum(dropdims(sum(inv_log; dims=dims); dims=dims) .> 0; dims=:scenarios);
+        dims=:scenarios,
+    )
+    return loc_count
+end
+
+"""
+    summed_inverse_rank(ranks::NamedDimsArray{D,T,3,A};dims::Union{Symbol,Vector{Symbol}}=[:scenarios, :timsteps],
+            ) where {D,T,A}
+    summed_inverse_rank(ranks::NamedDimsArray{D,T,2,A};) where {D,T,A}
+    summed_inverse_rank(ranks::NamedDimsArray,dims::Union{Symbol,Vector{Symbol}},)
+
+Calculates (number of sites) .- ranks summed over the dimension dims and transformed using agg_func 
+    (default no transformation).
+
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+- `dims` : Dimensions to sum over.
+
+# Returns 
+Inverse rankings (i.e. the greater the number the higher ranked the site).
+"""
+function summed_inverse_rank(
+    ranks::NamedDimsArray{D,T,3,A};
+    dims::Vector{Symbol}=[:scenarios, :timesteps],
+) where {D,T,A}
+    return summed_inverse_rank(ranks, dims)
+end
+function summed_inverse_rank(
+    ranks::NamedDimsArray{D,T,2,A};
+    dims::Vector{Symbol}=[:scenarios],
+) where {D,T,A}
+    return summed_inverse_rank(ranks, dims)
+end
+function summed_inverse_rank(
+    ranks::NamedDimsArray,
+    dims::Vector{Symbol},
+)
+    n_ranks = maximum(ranks)
+    inv_ranks = dropdims(sum(n_ranks .- ranks; dims=dims); dims=dims[1])
+    return inv_ranks ./ (n_ranks * prod([size(ranks, d) for d in dims]))
```

Comment:
You can replace `prod([size(ranks, d) for d in dims])` with `prod(dim(ranks, dims))`

Reply:
@Zapiano the function `dim()` is not recognised - is it from base or a package?

Reply:
Worked it out, it's from NamedDims

Reply:
I don't think it does the same thing though, for example for a set of 8 scenarios
`[size(ranks, d) for d in [:scenarios]] = 8`
whereas
` prod(NamedDims.dim(ranks, [:scenarios])) = 2`

Am I using this correctly?

Reply:
Yes you are, I just mixed up `dim`s with `axes`. What I meant was  `axes(ranks)` but I guess that won't work well in this case because you can't pass an array of `axis` as argument ... So I think you should just stick to your first solution. :)